Classes/Controller/AbstractSearchController.php

```diff
@@ -25,6 +25,23 @@
     // search matches
     private static $matches = ['title', 'abstract', 'author', 'language', 'tag', 'corporation', 'doctype', 'collections'];
 
+
+    protected function initializeView(\TYPO3\CMS\Extbase\Mvc\View\ViewInterface $view)
+    {
+        parent::initializeView($view);
+
+        if (TYPO3_MODE === 'BE') {
+            $selectedPageId = (int) \TYPO3\CMS\Core\Utility\GeneralUtility::_GP('id');
+            if ($selectedPageId) {
+                $client = $this->clientRepository->findAll()->current();
```

Comment:
@albig: What was the issue with using `findAll()->current();` again?

Reply:
`findAll()->current()` and `findAll()->getFirst()` are almost the same [according to TYPO3 API](https://api.typo3.org/typo3cms/7/html/class_t_y_p_o3_1_1_c_m_s_1_1_extbase_1_1_persistence_1_1_generic_1_1_query_result.html).

There is more logic checking for the query result is of the proper type. PHP and types! Anyway, it works. Especially because `findFirst()` is calling `current()` in the end.  :see_no_evil: